### ISSUE TITLE
match is erroring when given a non regex value

### ISSUE THREAD

**Issue by scottdavis** [cypress-io/cypress#58]
works: `.get('.scroll').should("have.prop", 'scrollLeft').and('match', /^0$/)`
errors: `.get('.scroll').should("have.prop", 'scrollLeft').and('match', 0)`

``` text
TypeError: t.exec is not a function
    at n.<anonymous> (http://localhost:2020/__cypress/static/js/vendor.js:239:8893)
    at n.require.register.n.exports.t.(anonymous function) (http://localhost:2020/__cypress/static/js/vendor.js:276:151)
    at n.<anonymous> (http://localhost:2020/__cypress/static/js/vendor.js:453:4980)
    at n.require.register.n.exports.t.(anonymous function) [as match] (http://localhost:2020/__cypress/static/js/vendor.js:422:122)
    at $Cypress.register.Cypress.addChildCommand.should.applyChainer (http://localhost:2020/__cypress/static/js/cypress.js:5417:34)
    at http://localhost:2020/__cypress/static/js/cypress.js:5448:24
    at Function.d.reduce.d.foldl.d.inject (http://localhost:2020/__cypress/static/js/vendor.js:27:8445)
    at http://localhost:2020/__cypress/static/js/cypress.js:5430:22
    at $Cypress.register.Cypress.addChildCommand.should (http://localhost:2020/__cypress/static/js/cypress.js:5453:32)
    at $Cy.<anonymous> (http://localhost:2020/__cypress/static/js/cypress.js:569:28)
    at $Cy.d.partial (http://localhost:2020/__cypress/static/js/vendor.js:27:14449)
    at $Cypress.extend.inject.$Cypress.Cy.sync.(anonymous function) (http://localhost:2020/__cypress/static/js/cypress.js:584:27)
    at $Cypress.register.Cypress.addChildCommand.and (http://localhost:2020/__cypress/static/js/cypress.js:5463:33)
    at $Cy.<anonymous> (http://localhost:2020/__cypress/static/js/cypress.js:569:28)
    at $Cy.d.partial (http://localhost:2020/__cypress/static/js/vendor.js:27:14449)
    at http://localhost:2020/__cypress/static/js/cypress.js:963:26
```

**Comment by brian-mann** [comment]
Yah this is actually coming from chai itself, it would be no different if you just wrote it out like so:

```javascript
expect("1000").to.match(0)
```

However you are not the first person to mention this (actually quite a few people have) and in the next release (next day or two) there will be several new error messages which handle this situation.

I'm not happy having to patch `chai` but it is what it is.

**Comment by scottdavis** [comment]
So the wisdom to take home here is always use regex?

**Comment by brian-mann** [comment]
Yes. Otherwise use "eq" or "deep.eq", etc. "match" is always for regex.

**Comment by brian-mann** [comment]
Fixed in [0.11.0](https://github.com/cypress-io/cypress/wiki/Changelog#0110-09132015).
